packages/devtools_app/lib/src/error_badge_manager.dart

```diff
@@ -44,16 +55,42 @@ class ErrorBadgeManager extends DisposableController
     if (e.extensionKind == 'Flutter.Error') {
       incrementBadgeCount(LoggingScreen.id);
 
-      final json = e.extensionData.data;
-      final objectId = json['objectId'] as String;
-      // TODO(kenz): verify that the object id is referring to an element or
-      // widget.
-      if (objectId?.contains('inspector-') ?? false) {
-        incrementBadgeCount(InspectorScreen.id);
+      final inspectableError = _extractInspectableError(e);
+      if (inspectableError != null) {
+        appendError(InspectorScreen.id, inspectableError);
       }
     }
   }
 
+  InspectableWidgetError _extractInspectableError(Event error) {
+    final node =
+        RemoteDiagnosticsNode(error.extensionData.data, null, false, null);
+
+    final errorSummaryNode = node.inlineProperties
+        ?.firstWhere((p) => p.type == 'ErrorSummary', orElse: () => null);
+    final errorMessage = errorSummaryNode?.description;
+    if (errorMessage == null) {
+      return null;
+    }
+
+    final devToolsUrlNode = node.inlineProperties?.firstWhere(
+      (p) =>
+          p.type == 'DevToolsDeepLinkProperty' &&
+          p.getStringMember('value') != null,
+      orElse: () => null,
+    );
+    if (devToolsUrlNode == null) {
+      return null;
+    }
+
+    final queryParams =
+        devToolsQueryParams(devToolsUrlNode.getStringMember('value'));
+    final inspectorRef =
+        queryParams != null ? queryParams['inspectorRef'] : null;
```

Comment:
Will it be possible to get the inspector ref for errors that do not have a deep link? @jacob314 @DanTup 

As is this would only work for overflow errors (the only errors with a deep link).

Reply:
When I first looked, the `inspectorRef` in the deep link didn't seem to be repeated anywhere else in the structured error, so I wasn't sure. If there is a way to get it (or add it - I'm not very familiar with what these IDs are or how they'd made), then I think this should be able to tweak this to work for others too.

Reply:
IIRC from when I was adding the deep link to overflow errors, the widget needed to be "registered" in a sense on the inspector service, which will happen when the inspector is initialized with all its widgets unless manually registered from Flutter (like in the case of widgets related to overflow errors). @jacob314 can add more context here.

Reply:
There will be multiple inspector refs for errors without a deep link but we shouldn't be showing them as we don't know what they reference. For example, if there are inspector refs for each widget up the parent tree, you don't want to highlight them all in red. Showing widgets in the tree in red is distracting so we want to be careful we don't create a bunch of false positives and users will start ignoring this feature.